### PR TITLE
Update message recovery indices check

### PR DESCRIPTION
**Pull request by Phanco**
### What was the problem?

This PR resolves #INSERT_ISSUE_NUMBER

### CODE REVIEW

File: framework/src/modules/interoperability/mainchain/commands/recover_message.ts [hunk 1]
```diff
@@ -113,6 +113,18 @@ export class RecoverMessageCommand extends BaseInteroperabilityCommand<Mainchain
 			};
 		}
 
+		// Check that the CCM indices do not exceed the outbox size. We check only the last one,
+		// as the idxs are sorted in ascending order.
+		if (
+			terminatedOutboxAccount.outboxSize <=
+			idxs[idxs.length - 1] - 2 ** terminatedOutboxAccount.outboxSize
```
Comment (AndreasKendziorra) [inline framework/src/modules/interoperability/mainchain/commands/recover_message.ts:120]: I assume that `2 ** terminatedOutboxAccount.outboxSize` is meant to be the prepended `1` to the binary representation of the index? If yes, then it must be `2**height(terminatedChainOutbox)` ... but of course we don't have `height(terminatedChainOutbox)`. Maybe it's easier to remove the most significant bit from `idxs[idxs.length - 1]` instead.

**Review by AndreasKendziorra — CHANGES_REQUESTED** [review]

**Review by ishantiw — APPROVED** [review]
LGTM

File: framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts [hunk 2]
```diff
@@ -276,6 +276,39 @@ describe('MessageRecoveryCommand', () => {
...
+					sendingChainID: utils.intToBuffer(2, 4),
+					receivingChainID: utils.intToBuffer(3, 4),
+					fee: BigInt(1),
+					status: CCMStatusCode.FAILED_CCM,
+					params: Buffer.alloc(0),
+				},
+			];
+			ccmsEncoded = ccms.map(ccm => codec.encode(ccmSchema, ccm));
+			transactionParams.crossChainMessages = [...ccmsEncoded];
+			transactionParams.idxs = [2 ** terminatedChainOutboxSize + terminatedChainOutboxSize];
```
Comment (AndreasKendziorra) [inline framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts:294]: This value must also be updated. You could use [getHeight](https://github.com/LiskHQ/lisk-sdk/blob/93d9291a0d8b8818fd7eb05c53520d988a26cbc3/elements/lisk-tree/src/merkle_tree/utils.ts#L48) to compute the edge case:
```
indexLength = getHeight(terminatedChainOutboxSize);
idxs = [terminatedChainOutboxSize + 2**indexLength]
```

**Review by AndreasKendziorra — CHANGES_REQUESTED** [review]

File: framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts (hunk 2 above)
Comment (Phanco) [reply framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts]: Updated, since the `getHeight` from merkle tree is not exposed, I copied the code to this file

File: framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts [hunk 3]
```diff
@@ -291,7 +293,9 @@ describe('MessageRecoveryCommand', () => {
 			];
 			ccmsEncoded = ccms.map(ccm => codec.encode(ccmSchema, ccm));
 			transactionParams.crossChainMessages = [...ccmsEncoded];
-			transactionParams.idxs = [2 ** terminatedChainOutboxSize + terminatedChainOutboxSize];
+
+			const indexLength = getHeight(terminatedChainOutboxSize);
+			transactionParams.idxs = [2 ** indexLength + terminatedChainOutboxSize];
```
Comment (AndreasKendziorra) [inline framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts:298]: As it is not straight forward to see what's happening here, you could leave a comment like this one:

"Choose the index such that the position of the ccm in the outbox tree is at terminatedChainOutboxSize, i.e. just outside of the tree. Note that a 1 has to be prepended to the binary representation of the index, which is done by adding 2**indexLength. See https://github.com/LiskHQ/lips/blob/main/proposals/lip-0031.md#proof-serialization."

Up to you.

**Review by AndreasKendziorra — APPROVED** [review]

File: framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts (hunk 3 above)
Comment (Phanco) [reply framework/test/unit/modules/interoperability/mainchain/commands/recover_message.spec.ts:298]: Sounds good, will add the above comment to `appendPrecedingToIndices` in this PR 👍 
https://github.com/LiskHQ/lisk-sdk/pull/8847
